db/seeds.d/95-mail_notifications.rb

```diff
@@ -24,8 +24,10 @@
     org_admin_role = Role.find_by(name: 'Organization admin')
 
     users = User.left_joins(:roles)
+                .joins(:auth_source)
                 .where(admin: true)
                 .or(User.where(id: UserRole.where(id: [org_admin_role.id] + org_admin_role.cloned_role_ids).select(:owner_id)))
+                .where.not(auth_source: { name: 'Hidden' })
```

Comment:
there are other potentially hiddden auth sources, e.g. https://github.com/theforeman/foreman_virt_who_configure/blob/master/app/models/foreman_virt_who_configure/auth_source_hidden_with_authentication.rb

I don't think it would cause problems though, since the user [is not](https://github.com/theforeman/foreman_virt_who_configure/blob/master/app/models/foreman_virt_who_configure/config.rb#L174-L192) admin, so that's just FYI

Reply:
Good to know, although even if we did set it up for hidden users, as long as they don't have mail enabled and email address set we would not attempt delivery to them anyway